lld/ELF/InputFiles.cpp

```diff
@@ -1418,6 +1423,67 @@ std::vector<uint32_t> SharedFile::parseVerneed(const ELFFile<ELFT> &obj,
   return verneeds;
 }
 
+// To determine if a shared file can support any of the GNU Attributes,
+// the .note.gnu.properties section need to be read. This has to be done
+// differently for SharedFiles as the information available is not as
+// extensive as a normal object input file. This will take the program
+// headers, along with the SHT_NOTE section header to find the relevant
+// information. This uses a similar process to the readGnuProperty
+// function, but designed specifically for SharedFiles.
+template <typename ELFT>
+void SharedFile::parseGnuAttributes(const uint8_t *base,
```

Comment:
I think `parseGnuAndFeatures` would be a better name than GnuAttributes as this is parsing GNU Properties for AndFeatures.